src/components/Editor/Preview/index.js

```diff
@@ -33,90 +33,102 @@ type State = {
-    }
-
-    if (!isPaused && wasPaused) {
-      codeMirror.off("tokenenter", this.onTokenEnter);
-      codeMirror.off("tokenleave", this.onTokenLeave);
-      codeMirrorWrapper.removeEventListener("mouseup", this.onMouseUp);
-      codeMirrorWrapper.removeEventListener("mousedown", this.onMouseDown);
-    }
+  componentWillUnmount() {
+    this.disposalble.dispose();
   }
 
-  updateHighlight(prevProps) {
-    const { preview } = this.props;
-
-    if (preview && !preview.updating) {
-      const target = getElementFromPos(preview.cursorPos);
-      target && target.classList.add("preview-selection");
-    }
-
-    if (prevProps.preview && !prevProps.preview.updating) {
-      const target = getElementFromPos(prevProps.preview.cursorPos);
-      target && target.classList.remove("preview-selection");
-    }
-  }
-
-  onTokenEnter = ({ target, tokenPos }) => {
-    this.props.updatePreview(target, tokenPos, this.props.editor.codeMirror);
-  };
-
-  onTokenLeave = e => {
-    if (!inPopup(e)) {
-      this.props.clearPreview();
+  // updateListeners(prevProps: ?Props) {
+  //   const { isPaused } = this.props;
+
+  //   const { codeMirror } = this.props.editor;
+  //   const codeMirrorWrapper = codeMirror.getWrapperElement();
+  //   const wasNotPaused = !prevProps || !prevProps.isPaused;
+  //   const wasPaused = prevProps && prevProps.isPaused;
+
+  //   if (isPaused && wasNotPaused) {
+  //     codeMirror.on("scroll", this.onScroll);
+  //     codeMirror.on("tokenenter", this.onTokenEnter);
+  //     codeMirror.on("tokenleave", this.onTokenLeave);
+  //     codeMirrorWrapper.addEventListener("mouseup", this.onMouseUp);
+  //     codeMirrorWrapper.addEventListener("mousedown", this.onMouseDown);
+  //   }
+
+  //   if (!isPaused && wasPaused) {
+  //     codeMirror.off("tokenenter", this.onTokenEnter);
+  //     codeMirror.off("tokenleave", this.onTokenLeave);
+  //     codeMirrorWrapper.removeEventListener("mouseup", this.onMouseUp);
+  //     codeMirrorWrapper.removeEventListener("mousedown", this.onMouseDown);
+  //   }
+  // }
+
+  // updateHighlight(prevProps) {
+  //   const { preview } = this.props;
+
+  //   if (preview && !preview.updating) {
+  //     const target = getElementFromPos(preview.cursorPos);
+  //     target && target.classList.add("preview-selection");
+  //   }
+
+  //   if (prevProps.preview && !prevProps.preview.updating) {
+  //     const target = getElementFromPos(prevProps.preview.cursorPos);
+  //     target && target.classList.remove("preview-selection");
+  //   }
+  // }
+
+  // onTokenEnter = ({ target, tokenPos }) => {
+  //   this.props.updatePreview(target, tokenPos, this.props.editor.codeMirror);
+  // };
+
+  // onTokenLeave = e => {
+  //   if (!inPopup(e)) {
+  //     this.props.clearPreview();
+  //   }
+  // }
+
+  onMouseOver = e => {
+    const { target } = e;
+    if (this.props.selectedFrameVisible) {
+      console.log("target");
+      this.props.updatePreview(target, this.props.editor);
```

Comment:
getting the preview popup to work well has been one of the biggest challenges. Can't wait to test that